x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStartTransformAction.java

```diff
@@ -205,7 +214,24 @@ protected void masterOperation(
                 return;
             }
             // Validate source and destination indices
-            SourceDestValidator.validate(config, clusterService.state(), indexNameExpressionResolver, false);
+            ValidationException validationResult = SourceDestValidator.validate(
+                clusterService.state(),
+                indexNameExpressionResolver,
+                transportService.getRemoteClusterService(),
+                isRemoteSearchEnabled
+                    ? new RemoteClusterLicenseChecker(client, XPackLicenseState::isTransformAllowedForOperationMode)
+                    : null,
+                config.getSource().getIndex(),
+                config.getDestination().getIndex(),
+                clusterService.getNodeName(),
+                "basic",
```

Comment:
```suggestion
            License.OperationMode.BASIC.description()
```
We should use the operation mode enum.